.github/workflows/system-install.yml

```diff
@@ -0,0 +1,115 @@
+name: System Install
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  RUSTUP_MAX_RETRIES: 10
+  PYTHON_VERSION: "3.12"
+
+jobs:
+  # Install Python on Ubuntu.
+  install-ubuntu:
+    name: "Install Python on Ubuntu"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: "Install Rust toolchain"
+        run: rustup update
```

Comment:
This update looks like it has some weird conflicts — I think we also don't do `update` elsewhere and just do `rustup show`?

Reply:
Oops, thanks.